java/google/registry/whois/DomainOrHostLookupCommand.java

```diff
@@ -52,17 +49,14 @@ public final WhoisResponse executeQuery(final DateTime now) throws WhoisExceptio
     }
     // Google Policy: Do not return records under TLDs for which we're not authoritative.
     if (tld.isPresent() && getTlds().contains(tld.get().toString())) {
-      T domainOrHost = loadByForeignKey(
-          new TypeInstantiator<T>(getClass()){}.getExactType(),
-          domainOrHostName.toString(),
-          now);
-      if (domainOrHost != null) {
-        return getSuccessResponse(domainOrHost, now);
+      final Optional<WhoisResponse> response = getSuccessResponse(domainOrHostName, now);
+      if (response.isPresent()) {
+        return response.get();
       }
     }
     throw new WhoisException(now, SC_NOT_FOUND, errorPrefix + " not found.");
   }
 
   /** Renders a response record, provided its successfully retrieved datastore entity. */
-  abstract WhoisResponse getSuccessResponse(T domainOrHost, DateTime now);
+  protected abstract Optional<WhoisResponse> getSuccessResponse(InternetDomainName domainName, DateTime now);
```

Comment:
This seems like a bit of a misleading name for the method now, since the response isn't necessarily "success" - it could be absent now, unlike before.  Javadoc is also stale.  I'd rename the method to something like getResponseIfFound() or even just getResponse() and update the javadoc.